Copilot: Add Vim to docs to make it clear it supports both Vim/Neovim

ychin opened the pull request:
### Why:

The repository for Copilot (Neo)Vim plugin was updated to support Vim a while ago. Update the docs to reflect that.

Repository link: https://github.com/github/copilot.vim

Closes: #24624 


### What's being changed (if available, include any code snippets, screenshots, or gifs):

~~I didn't rename the files for now, as it didn't seem important enough.~~ Turned out I need to rename the files to match the titles to pass tests. Ran `script/reconcile-filenames-with-ids.js` to rename the files.

Just changed the title and texts ("Neovim" → "Vim/Neovim"), and added Vim-specific information (mininum version and correct installation procedures pulled from the [repo's page](https://github.com/github/copilot.vim)).

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

AS0820075980 commented:
Chaiwat oppo A16

ในวันที่ พฤ. 23 มี.ค. 2023 18:41 น. ชัยวัฒน์ จันทร์ทักษ์ <
***@***.***> เขียนว่า:

>
>
> Chaiwat oppo A16
>
> ในวันที่ พฤ. 23 มี.ค. 2023 18:38 น. github-actions[bot] <
> ***@***.***> เขียนว่า:
>
>> Automatically generated comment ℹ️
>>
>> *This comment is automatically generated and will be overwritten every
>> time changes are committed to this branch.*
>>
>> The table contains an overview of files in the content directory that
>> have been changed in this pull request. It's provided to make it easy to
>> review your changes on the staging site. Please note that changes to the
>> data directory will not show up in this table.
>> ------------------------------
>> Content directory changes
>>
>> *You may find it useful to copy this table into the pull request summary.
>> There you can edit it to share links to important articles or changes and
>> to give a high-level overview of how the changes in your pull request
>> support the overall goals of the pull request.*
>> *Source* *Preview* *Production* *What Changed*
>> copilot/configuring-github-copilot/configuring-github-copilot-in-neovim.md
>> <https://github.com/github/docs/blob/9ee027a12a78f71328900a081549903b4d5c74da/content%2Fcopilot%2Fconfiguring-github-copilot%2Fconfiguring-github-copilot-in-neovim.md>
>> fpt
>> <https://docs-24625-e61529.preview.ghdocs.com/copilot/configuring-github-copilot/configuring-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/configuring-github-copilot/configuring-github-copilot-in-neovim>
>> fpt
>> <https://docs.github.com/copilot/configuring-github-copilot/configuring-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/configuring-github-copilot/configuring-github-copilot-in-neovim>
>>
>> copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim.md
>> <https://github.com/github/docs/blob/9ee027a12a78f71328900a081549903b4d5c74da/content%2Fcopilot%2Fgetting-started-with-github-copilot%2Fgetting-started-with-github-copilot-in-neovim.md>
>> fpt
>> <https://docs-24625-e61529.preview.ghdocs.com/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim>
>> fpt
>> <https://docs.github.com/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim>
>> copilot/overview-of-github-copilot/about-github-copilot-for-individuals.md
>> <https://github.com/github/docs/blob/9ee027a12a78f71328900a081549903b4d5c74da/content%2Fcopilot%2Foverview-of-github-copilot%2Fabout-github-copilot-for-individuals.md>
>> fpt
>> <https://docs-24625-e61529.preview.ghdocs.com/copilot/overview-of-github-copilot/about-github-copilot-for-individuals>
>> ghec
>> ***@***.***/copilot/overview-of-github-copilot/about-github-copilot-for-individuals>
>> fpt
>> <https://docs.github.com/copilot/overview-of-github-copilot/about-github-copilot-for-individuals>
>> ghec
>> ***@***.***/copilot/overview-of-github-copilot/about-github-copilot-for-individuals>
>>
>> copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim.md
>> <https://github.com/github/docs/blob/9ee027a12a78f71328900a081549903b4d5c74da/content%2Fcopilot%2Ftroubleshooting-github-copilot%2Ftroubleshooting-github-copilot-in-neovim.md>
>> fpt
>> <https://docs-24625-e61529.preview.ghdocs.com/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim>
>> fpt
>> <https://docs.github.com/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim>
>> ghec
>> ***@***.***/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim>
>> ------------------------------
>>
>> fpt: Free, Pro, Team
>> ghec: GitHub Enterprise Cloud
>> ghes: GitHub Enterprise Server
>> ghae: GitHub AE
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/github/docs/pull/24625#issuecomment-1481031286>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A43CNFP5ROZ7ECKEQDN6YQLW5QY4DANCNFSM6AAAAAAWFCBIAU>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>

AS0820075980 commented:
Chaiwat oppo A16

ในวันที่ พฤ. 23 มี.ค. 2023 19:10 น. Yee Cheng Chin ***@***.***>
เขียนว่า:

> Ok, tests are passing now. Had to rename the files due to a title change
> hence slightly more files than I was hoping for, but the commits are
> separate to make it easier to see what changed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/github/docs/pull/24625#issuecomment-1481083160>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43CNFMHFO7LNNGS2IOWCU3W5Q4VHANCNFSM6AAAAAAWFCBIAU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

cmwilson21 commented:
@ychin Thank you for opening a PR and linking it to your issue! ✨ 

I'll get this triaged for review! :zap:

ychin commented:
Hi! I'm curious if there is a timeline for review? I'm now getting merge conflicts. I can resolve them but just wanted to see if there will be a reviewer scheduled to take a look. Thanks.

cmwilson21 commented:
@ychin Thanks for checking in on this. We are working through our backlog, so we appreciate your patience ❤️ 

Someone will review it soon! 👍

ychin committed:
Copilot: Add Vim to docs to make it clear it supports both Vim/Neovim

The repository for Copilot (Neo)Vim plugin was updated to support Vim a
while ago. Update the docs to reflect that.

hectorsector commented on lib/all-tools.js:
```diff
@@ -9,7 +9,7 @@ export const allTools = {
   javascript: 'JavaScript',
   jetbrains: 'JetBrains IDEs (Beta)',
-  neovim: 'Neovim',
+  vimneovim: 'Vim/Neovim',
```
We're good on the addition of Vim/Neovim as a tool for the picker from an Engineering POV. However, I think we'll still need to get a review on the Copilot-specific content changes @cmwilson21

hectorsector replied:
Thank you for getting this opened @ychin!

ychin replied:
I'm not really sure who's responsibility it is to do that? Should I ping someone to do so or someone on your team will do so?

ychin commented:
@cmwilson21 Per the comment above, I think I still need someone from Copilot to take a look per @hectorsector 's comment?

cmwilson21 commented:
@ychin Thanks for checking in! This PR is on the board and waiting for review. We appreciate your patience as we work through some of our backlog. Someone will have eyes on it soon! 💖

jules-p reviewed (approved):
Hey @ychin :wave: thank you for opening this PR, it looks great :sparkles: I'll get these changes merged for you.

jules-p committed:
Merge branch 'main' into add-vim-to-copilot-docs
